src/snowflake/snowpark/ml/transformer.py

```diff
@@ -0,0 +1,501 @@
+        X_std = (X - X.min(axis=0)) / (X.max(axis=0) - X.min(axis=0))
+        X_scaled = X_std * (max - min) + min
+        where min, max are decided by feature_range
+        """
+        if not self._fit_was_called:
+            raise Exception(
+                "The transformer was not fitted yet, call fit() function before transform"
+            )
+        elif not self._output_cols or len(self._input_cols) != len(self._output_cols):
+            raise ValueError(
+                "The number of input column and output column does not match or output column is empty"
+            )
+        states_table = df._session.table(self._states_table_name)
+
+        res_column = df.columns
+
+        for input_col, states_col, output_col in zip(
+            self.input_cols, self._states_table_cols, self.output_cols
+        ):
+            res_column.append(
+                (
+                    (df[input_col] - col(states_col)["min"])
+                    / iff(
+                        (col(states_col)["max"] - col(states_col)["min"]) == 0,
+                        1,
+                        col(states_col)["max"] - col(states_col)["min"],
+                    )
+                    * (self.maximum - self.minimum)
+                    + self.minimum
+                ).as_(output_col)
+            )
+
+        df_transform = states_table.join(df).select(res_column)
+        return df_transform
+
+
+class OneHotEncoder(Transformer):
+    def __init__(
+        self,
+        input_cols: Optional[List[str]] = None,
+        output_cols: Optional[List[str]] = None,
+        category: Union[str, List[List[str]]] = "auto",
+    ):
+        super().__init__(input_cols, output_cols)
+        self._category = category
+        self._merge = False
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{input_col}_encoder" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._category_table_name = {}
+        self._encoder_count_table = random_name_for_temp_object(TempObjectType.TABLE)
+
+    @Transformer.input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self._encoder_table_cols = (
+            [f"states_{input_col}_encoder" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    def fit(self, df: DataFrame) -> "OneHotEncoder":
+        super().fit(df)
+
+        encoder_fit(self, df, "onehot")
+        return self
+
+    def transform(
+        self,
+        df: DataFrame,
+        dense: bool = True,
```

Comment:
sorry I meant just changing the variable and column name in the code, but not changing the argument name. We should keep it as `sparse`.